appserver/static/javascript/views/setup_view.js

```diff
@@ -0,0 +1,626 @@
+            },
+
+            // ---------------------
+            // Creation Functions
+            // ---------------------
+            create_splunk_js_sdk_service: function create_splunk_js_sdk_service(
+                splunk_js_sdk,
+                application_name_space,
+            ) {
+                var http = new splunk_js_sdk.SplunkWebHttp();
+
+                var splunk_js_sdk_service = new splunk_js_sdk.Service(
+                    http,
+                    application_name_space,
+                );
+
+                return splunk_js_sdk_service;
+            },
+
+            create_configuration_file: function create_configuration_file(
+                configurations_accessor,
+                configuration_file_name,
+            ) {
+                var parent_context = this;
+
+                return configurations_accessor.create(configuration_file_name, function (
+                    error_response,
+                    created_file,
+                ) {
+                    // Do nothing
+                });
+            },
+
+            create_stanza: function create_stanza(
+                configuration_file_accessor,
+                new_stanza_name,
+            ) {
+                var parent_context = this;
+
+                return configuration_file_accessor.create(new_stanza_name, function (
+                    error_response,
+                    created_stanza,
+                ) {
+                    // Do nothing
+                });
+            },
+
+            update_stanza_properties: function update_stanza_properties(
+                configuration_stanza_accessor,
+                new_stanza_properties,
+            ) {
+                var parent_context = this;
+
+                return configuration_stanza_accessor.update(
+                    new_stanza_properties,
+                    function (error_response, entity) {
+                        // Do nothing
+                    },
+                );
+            },
+
+            // ----------------------------------
+            // Input Cleaning and Checking
+            // ----------------------------------
+            sanitize_string: function sanitize_string(string_to_sanitize) {
+                var sanitized_string = string_to_sanitize.trim();
+
+                return sanitized_string;
+            },
+
+            validate_api_url_input: function validate_api_url_input(hostname) {
+                var error_messages = [];
+
+                var is_string_empty = typeof hostname === "undefined" || hostname === "";
+                var does_string_start_with_http_protocol = hostname.startsWith("http://");
+                var does_string_start_with_https_protocol = hostname.startsWith(
+                    "https://",
+                );
+
+                if (is_string_empty) {
+                    error_message =
+                        "The `API URL` specified was empty. Please provide" + " a value.";
```

Comment:
minor: We're calling the field "AMP for Endpoints Console Hostname" not "API URL"